include/unix/osd.h

```diff
@@ -59,12 +61,19 @@ static inline void ofi_freealign(void *memptr)
 
 static inline ssize_t ofi_read_socket(int fd, void *buf, size_t count)
 {
-	return read(fd, buf, count);
+	return recv(fd, buf, count, 0);
 }
 
 static inline ssize_t ofi_write_socket(int fd, const void *buf, size_t count)
 {
-	return write(fd, buf, count);
+	int flags = 0;
+
+	/* MSG_NOSIGNAL doesn't exists on OS X */
+#if !defined __APPLE__
+	flags = MSG_NOSIGNAL;
+#endif
```

Comment:
Thanks. As you suggested, I have included a function ofi_send_socket(), which accepts flags.
The function has been added into `include/unix/osd.h` and `include/windows/osd.h`.
I modified the socket provider in a later patch to prevent the application to emit a SIGBUS signal after disconnect.